### PR TITLE
drivers/hdc1000: (re)add error checks for I2C access

### PR DESCRIPTION
**Pull request by haukepetersen**
As pointed out in #7008, it seems I was a little bit to optimistic w.r.t. the (long-term) reliability of the I2C bus in general. Hence error checking for every I2C transaction is actually needed, so (re)adding it here.

### CODE REVIEW

**Comment by immesys** [comment]
Awesome! Thanks for doing this

**Comment by immesys** [comment]
This actually raises a question for us, we are trying to move our applications to using the SAUL interface, but many of our sensors do multiple things at once (e.g hdc1000 can sample rh+temp at the same time, tmp006 does die temp, voltage and maybe object temp). It is inefficient to sample the underlying sensor twice when reading the temp + humidity via SAUL, but we don't know how to return multiple results with different units via the SAUL interface (it seems that multimensional results are ok but only if they all have the same unit). How should we tackle this?

**Comment by Hyungsin** [comment]
Good for me too.

@haukepetersen, another issue I have in my mind. 
SAUL abstraction does not support to read both temperature and humidity together.
It requires to make two separate SAUL drivers, one for temperature and the other for humidity.

The thing is that currently, the hdc1000 driver samples the two information together, which requires more time than reading just one information. So, when a SAUL temp driver tries to read a temp, hdc1000 will read temp and humidity together. When a SAUL humidity driver tries to read humidity, hdc1000 will read the two info again.

I agree that sampling the two information together is more efficient, but given that we cannot do that through SAUL abstraction, hdc1000 driver implementation needs to be investigated in the perspective of interaction with SAUL.

File: drivers/hdc1000/hdc1000.c [hunk 1]
```diff
@@ -86,36 +87,48 @@ void hdc1000_trigger_conversion(hdc1000_t *dev)
      * to the address 0x00 (HDC1000_TEMPERATURE).
      * Conversion Time is 6.50ms for each value for 14 bit resolution.
      */
-    i2c_write_byte(dev->p.i2c, dev->p.addr, HDC1000_TEMPERATURE);
+    if (i2c_write_byte(dev->p.i2c, dev->p.addr, HDC1000_TEMPERATURE) != 1) {
+        status = HDC1000_BUSERR;
```
Comment (smlng) [inline drivers/hdc1000/hdc1000.c]: why not simply `return HDC100_BUSERR`? Omitting `int status`, though I think compilers will optimize that away anyway.

File: drivers/hdc1000/hdc1000.c [hunk 2]
```diff
@@ -86,36 +87,48 @@ void hdc1000_trigger_conversion(hdc1000_t *dev)
...
+    }
 
     i2c_release(dev->p.i2c);
+    return status;
```
Comment (smlng) [inline drivers/hdc1000/hdc1000.c]: and `return HDC1000_OK` here

File: drivers/hdc1000/hdc1000.c (hunk 1 above)
Comment (immesys) [inline drivers/hdc1000/hdc1000.c]: Must unlock i2c mutex before return

Comment (smlng) [inline drivers/hdc1000/hdc1000.c]: ah right, oversaw that first ...

File: drivers/hdc1000/hdc1000.c (hunk 2 above)
Comment (smlng) [inline drivers/hdc1000/hdc1000.c]: nevermind, see above

**Comment by haukepetersen** [comment]
@immesys For sensors that have different 'sources' of data, I propose the following approach: whenever one of the values is read, the SAUL layer reads all values, returns the selected one, but buffers all other values. Now we have to options: we define one of the values as 'master', so everytime the master value is read, the buffers are updated. This way we have time to read the 'slave' values until we read the master value again. Alternatively we can put a timeout on the buffered values (-> this I have done for the `dht` sensor once: `drivers/dht/dht_saul.c`). Also a combination of both approaches may be possible. I guess it is up to the nature of the sensor (e.g. sampling rate, synchronization of sources) which solution fits best...

Note: just noticed that the implementation for the DHT sensor is kind of broken, it will break if more than 1 DHT sensor is initialized and used via SAUL... The buffers should be moved into the device descriptor...

**Comment by haukepetersen** [comment]
ACKed and Murdock(2) is happy -> go

**Pull request merged**